ravenml/data/interfaces.py

```diff
@@ -23,6 +34,98 @@ class CreateInput(object):
+            self.dataset_path = dp
+        
+        ## Set up Imageset
+        # prompt for dataset if not provided
+        imageset_list = config.get('imageset')
+        if imageset_list is None:
+            imageset_options = cli_spinner('No imageset provided. Finding imagesets on S3...', get_imageset_names)
+            imageset_list = user_selects('Choose imageset:', imageset_options, selection_type="checkbox")
+        else: 
+            for imageset in imageset_list:
+                if imageset not in get_imageset_names():
+                    hint = 'imageset name, no such imageset exists on S3'
+                    raise click.exceptions.BadParameter(imageset_list, param=imageset_list, param_hint=hint)  
+    
+        ## Set up Basic Metadata
+        # TODO: add environment description, git hash, etc
+        self.metadata = config.get('metadata', {})
+        # handle user defined metadata fields
+        if not self.metadata.get('created_by'):
+            self.metadata['created_by'] = user_input('Please enter your first and last name:')
+        if not self.metadata.get('comments'):
+            self.metadata['comments'] = user_input('Please enter descriptive comments about this training:')
+        
+        if config.get('kfolds'):
+            self.metadata['kfolds'] = config['kfolds']
+        if config.get('test_percent'):
+            self.metadata['test_percent'] = config['test_percent']
+        if config.get('filter'):
+            self.metadata['filter'] = config['filter']
+        else:
+            self.metadata['filter'] = False
+
+        # Initialize Directory for Dataset    
+        self.metadata['dataset_name'] = config['dataset_name'] if config.get('dataset_name') else user_input(message="What would you like to name this dataset?")
+        os.makedirs(dp / self.metadata['dataset_name'], exist_ok=True)  
+
+        # handle automatic metadata fields
+        self.metadata['date_started_at'] = datetime.utcnow().isoformat() + "Z"
+        self.metadata['imagesets_used'] = imageset_list
+        
+        ## Set up fields for plugin use
+        # NOTE: plugins should overwrite the architecture field to something
+        # more specific/useful since it is used to name the final uploaded model
+        self.metadata[plugin_name] = {'architecture': plugin_name, 'temp_dir_path': self.plugin_cache.path / TEMP_DIR_PREFIX}
+        # plugins should only ACCESS the plugin_metadata attibute and add items. They should
+        # NEVER assign to the attribute as it will break the reference to the overall metadata dict
+        self.plugin_metadata = self.metadata[plugin_name]
+        if not config.get('plugin'):
+            raise click.exceptions.BadParameter(config, param=config, param_hint='config, no "plugin" field. Config was')
+        else:
+            self.plugin_config = config.get('plugin')
+
+class CreateOutput(object):
+    
+    def __init__(self, create: CreateInput):
```

Comment:
temp_dir is the only field that should actually be an attribute of this class. all the other attributes can be obtained directly in the `process_result` function back in the data `commands.py`, in the same way they are found for the train command, by accessing the `CreateInput` which is still located at `ctx.obj`.

That way, the output is clear about exactly what the plugin must provide back to the core - in this case, just the temp directory.